packages/server/src/fhir/repo.ts

```diff
@@ -2344,6 +2347,15 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
       } finally {
         this.endTransaction();
       }
+
+      const baseDelayMs = config.transactionExpBackoffBaseDelayMs ?? defaultExpBackoffBaseDelayMs;
+      // Attempts are 0-indexed, so first wait after first attempt will be somewhere between 75% and 100% of baseDelayMs
+      // This calculation results in something like this for the default values:
+      // Attempt 1: 50 * (2^0) = 50 * [0.75, 1] = **[37.5, 50] ms**
+      // Attempt 2: 50 * (2^1) = 100 * [0.75, 1] = **[75, 100] ms**
+      // etc...
+      const delayMs = Math.ceil(baseDelayMs * 2 ** attempt * (0.75 + Math.random() * 0.25));
+      await sleep(delayMs);
```

Comment:
We should guard here, otherwise we will perform an unnecessary `sleep` on the final iteration.  It's not the end-of-the-world, but might be nice to add a guard.

Reply:
Done